### PR TITLE
Split filter! docs, add # Examples, add doctests, move from helpdb

### PR DESCRIPTION
**Pull request by kshyatt**
Seems like it's better to have two docs for `filter!` if the `Associative` one needs a function that takes two arguments.

### CODE REVIEW

**Review by ararslan — APPROVED** [review]

File: base/array.jl [hunk 1]
```diff
@@ -1958,6 +1958,23 @@ Dict{Int64,String} with 1 entry:
 """
 filter(f, As::AbstractArray) = As[map(f, As)::AbstractArray{Bool}]
 
+"""
+    filter!(function, a::AbstractVector)
```
Comment (tkelman) [inline base/array.jl]: function is a keyword, should use a different name

File: base/array.jl [hunk 2]
```diff
@@ -1958,6 +1958,23 @@ Dict{Int64,String} with 1 entry:
...
+
+Update `collection`, removing elements for which `function` is `false`.
```
Comment (tkelman) [inline base/array.jl]: collection isn't in the signature

**Comment by kshyatt** [comment]
I ended up fixing `filter` too, which had many of the same problems. Better, @tkelman ?

File: base/associative.jl [hunk 3]
```diff
@@ -327,6 +327,24 @@ function filter!(f, d::Associative)
...
     return d
 end
+
+"""
+    filter(f, d::Associative)
+
+Return a copy of `d`, removing elements for which `f` is `false`.The function `f` is passed two arguments (key and value).
+
+# Examples
+```jldocttest
```
Comment (tkelman) [inline base/associative.jl]: misspelled

File: base/array.jl [hunk 4]
```diff
@@ -1928,10 +1928,10 @@ end
...
-    filter(function, collection)
+    filter(f, a::AbstractArray)
 
-Return a copy of `collection`, removing elements for which `function` is `false`. For
-associative collections, the function is passed two arguments (key and value).
+Return a copy of `a`, removing elements for which `f` is `false`.
+The function `f` is passed one argument.
 
 # Examples
 ```jldocttest
```
Comment (tkelman) [inline base/array.jl]: misspelled

File: base/associative.jl [hunk 5]
```diff
@@ -327,6 +327,24 @@ function filter!(f, d::Associative)
     end
     return d
 end
+
+"""
+    filter(f, d::Associative)
+
+Return a copy of `d`, removing elements for which `f` is `false`.The function `f` is passed two arguments (key and value).
```
Comment (KristofferC) [inline base/associative.jl]: Also, space between `.` and `The`.

**Comment by kshyatt** [comment]
OK now?

File: base/array.jl [hunk 6]
```diff
@@ -1945,19 +1945,27 @@ julia> filter(isodd, a)
...
 """
 filter(f, As::AbstractArray) = As[map(f, As)::AbstractArray{Bool}]
 
+"""
+    filter!(f, a::AbstractVector)
+
+Update `a`, removing elements for which `function` is `false`.
+The function `f` is passed one argument.
+
+# Example
```
Comment (tkelman) [inline base/array.jl]: are we going to move to Examples plural everywhere?

File: base/array.jl [hunk 7]
```diff
@@ -1945,19 +1945,27 @@ julia> filter(isodd, a)
...
-Dict{Int64,String} with 1 entry:
-  1 => "a"
 ```
 """
 filter(f, As::AbstractArray) = As[map(f, As)::AbstractArray{Bool}]
 
+"""
+    filter!(f, a::AbstractVector)
+
+Update `a`, removing elements for which `function` is `false`.
```
Comment (Sacha0) [inline base/array.jl]: `function` -> `f`?

**Review by Sacha0 — APPROVED** [review]
lgtm modulo the typo! :)

**Comment by kshyatt** [comment]
I think I got everything!

**Pull request merged**